sui_programmability/framework/sources/Governance/ValidatorSet.move

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module Sui::ValidatorSet {
+    use Std::Vector;
+
+    use Sui::Coin::Coin;
+    use Sui::SUI::SUI;
+    use Sui::Transfer;
+    use Sui::TxContext::{Self, TxContext};
+    use Sui::Validator::{Self, Validator};
+
+    friend Sui::SuiSystem;
+
+    const EDUPLICATE_VALIDATOR: u64 = 0;
+
+    const EINVALID_VALIDATOR_INDEX: u64 = 1;
+
+    const EVALIDATOR_NOT_FOUND: u64 = 2;
+
+    const EDUPLICATE_WITHDRAW: u64 = 3;
+
+    const ENOT_ENOUGH_VALIDATORS: u64 = 4;
+
+    struct ValidatorSet has store {
```

Comment:
I think the total stake and quorum threshold (i.e., 66% of the total stake) would also be useful to store here. Certificate checking logic will need to know these numbers in addition to the validator set.